src/script/interpreter.cpp

```diff
@@ -1106,8 +1110,64 @@ class CTransactionSignatureSerializer {
 
 } // anon namespace
 
-uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType)
+uint256 SignatureHash(const CScript& scriptCode, const CTransaction& txTo, unsigned int nIn, int nHashType, const CAmount& amount, int sigversion)
 {
+    if (sigversion == 1) {
+        uint256 hashPrevouts;
+        uint256 hashSequence;
+        uint256 hashOutputs;
+
+        if (!(nHashType & SIGHASH_ANYONECANPAY)) {
+            CHashWriter ss(SER_GETHASH, 0);
+            for (unsigned int n = 0; n < txTo.vin.size(); n++) {
+                ss << txTo.vin[n].prevout;
+            }
+            hashPrevouts = ss.GetHash(); // TODO: cache this value for all signatures in a transaction
+        }
+
+        if (!(nHashType & SIGHASH_ANYONECANPAY) && (nHashType & 0x1f) != SIGHASH_SINGLE && (nHashType & 0x1f) != SIGHASH_NONE) {
```

Comment:
> If you don't care all outputs, other people may want to update their outputs, for example, with opt-in RBF (or to disable opt-in RBF to make the tx final)

RBF only matters once the tx is propagated and the tx has to be "final" and "BIP68 final" before it'll propagate.
Before it's broadcast, they can freely change their outputs since a SINGLE or NONE signature allows that.
Once a tx gets propagated that opts-out of RBF, then they wouldn't be able to change their sequence to opt-in and be able to get the new tx relayed.

Since I care about all the inputs, I think I'd want to have to sign them changing to finalize early if that's different than what I was expecting.
If I anticipated them using a final sequence number (because that's what the protocol defined or what the parties agreed upon), then I'd just sign that to begin with.

Same with the BIP68 stuff. It seems mostly benign (the first case you mentioned is a potential stumbling point that a protocol might have to account for), but I haven't seen a compelling reason to give other parties the flexibility of changing sequence numbers when I care about all the inputs.

I'm thinking that the proper sequence number to sign would be defined by the protocol or coordinated based on the parties' preferences for locktime enablement and replaceability. I think that's reasonable, but maybe I'm wrong. Eh, it probably doesn't matter. I was just curious.